source/numerics/time_dependent.cc

```diff
@@ -863,7 +864,7 @@ void TimeStepBase_Tria<dim>::refine_grid (const RefinementData refinement_data)
           if (cell->refine_flag_set())
             previous_cells += (GeometryInfo<dim>::max_children_per_cell-1);
           else if (cell->coarsen_flag_set())
-            previous_cells -= (GeometryInfo<dim>::max_children_per_cell-1) /
+            previous_cells -= (double)(GeometryInfo<dim>::max_children_per_cell-1) /
                               GeometryInfo<dim>::max_children_per_cell;
```

Comment:
OK, thanks for confirming. Feel free to add a note to the changes.h file if you want, otherwise just merge once the tester is through.